src/app/features/public-form/public-form.component.ts

```diff
@@ -93,10 +98,20 @@ export class PublicFormComponent<E extends Entity> implements OnInit {
     this.formConfig = publicForms.find(
       (form: PublicFormConfig) => form.route === id || form.getId(true) === id,
     );
+    if (!this.formConfig) {
+      this.publicFormNotFound = true;
+      return;
+    }
 
     this.entityType = this.entities.get(
       this.formConfig.entity,
     ) as EntityConstructor<E>;
+    if (this.ability.cannot("create", this.entityType)) {
+      this.alertService.addDanger(
+        "You do not have the required permissions to submit this form. Please log in or contact your system administrator",
+      );
```

Comment:
let's make this a text like the "not found" in the template also. Alerts tend to be easily overlooked by users.

(maybe turn the "publicFormNotFound" into a general "error" property that can hold different values?)

Reply:
sure, let me make these changes

Reply:
Heading is wrong in this case (claiming "form not found"). If the user has permission to the form config but not to create a new record, we can display the actual form title and description, I think, and only below that display the permission error. Then the user has more context?

Test Page: https://dev.aam-digital.net/public-form/schoolform

What do you think from a user's perspective, @george-neha ?

Reply:
As a user, the message should be empathetic and should also indicate the next/other steps the user could take. 
The wording for the message could be :

**Access Denied to Public Form-Title**

Unfortunately, access has been denied due to insufficient permissions to access this Public Form.
Please contact your team for further assistance.

Reply:
@george-neha it's looking better now?
<img width="1474" alt="image" src="https://github.com/user-attachments/assets/8ae8c4fe-40e7-44e0-a734-04c510d17686" />